app/k9mail/src/main/java/com/fsck/k9/account/AccountLoader.kt

```diff
@@ -0,0 +1,58 @@
+package com.fsck.k9.account
+
+import app.k9mail.feature.account.common.AccountCommonExternalContract
+import app.k9mail.feature.account.common.domain.entity.Account
+import app.k9mail.feature.account.common.domain.entity.AccountOptions
+import com.fsck.k9.Preferences
+import com.fsck.k9.logging.Timber
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import com.fsck.k9.Account as K9Account
+
+class AccountLoader(
+    private val preferences: Preferences,
```

Comment:
We could use the `AccountManager` interface instead of `Preferences`. That would make it easier to add tests for this class.

Reply:
Yeah, it's a bit hidden but makes sense.